src/plugins/graph/models/graph-model.ts

```diff
@@ -470,9 +472,27 @@ export const GraphModel = TileContentModel
       const colorIndex = self._idColors.get(id);
       if (colorIndex === undefined) return "black";
       return clueGraphColors[colorIndex % clueGraphColors.length].name;
+    },
+    get isShowingMovableLine() {
+      return self.adornments.find(a => a.type === kMovableLineType)?.isVisible;
     }
   }))
   .actions(self => ({
+    showMovableLine() {
+      const mLine = MovableLineModel.create();
+      mLine.setInitialLine(self.axes.get("bottom"), self.axes.get("left"), "{}");
+      self.showAdornment(mLine);
+    },
+    hideMovableLine() {
+      self.hideAdornment(kMovableLineType);
+    },
+    toggleMovableLine() {
+      if (self.isShowingMovableLine) {
+        this.hideMovableLine();
+      } else {
+        this.showMovableLine();
+      }
+    },
```

Comment:
A goal of the adornments infrastructure and, in particular, the adornment registration system is to avoid the need for this kind of adornment-specific coding. If you look at the implementation of CODAP's `AdornmentCheckbox` class in `adornment-checkbox.tsx`, a single checkbox component can be used to add/hide/show many different adornments by passing in the appropriate adornment type and looking up everything else it needs from the registration system. The `AdornmentCheckbox` component itself probably won't be useful to you since the CLUE UI is different, but the principles should still apply.

All of which is to say that you don't really need these movable-line-specific methods here, but if you'd rather add them for expediency I leave that to your discretion.

Reply:
Good point.  I did some refactoring to organize this code better.  Connecting lines is still referred to in the graph-model, but that's a different story.